src/javascript/app_2/Modules/Trading/Components/Form/Purchase/contract_info.jsx

```diff
@@ -3,60 +3,63 @@ import React             from 'react';
 import { localize }      from '_common/localize';
 import Money             from 'App/Components/Elements/money.jsx';
 import Tooltip           from 'App/Components/Elements/tooltip.jsx';
-import { IconTradeType } from 'Assets/Trading/Types';
+import { IconPriceMove } from 'Assets/Trading/icon_price_move.jsx';
 
-const ContractInfo = ({
-    barrier_count,
-    contract_title,
-    contract_type,
-    currency,
-    proposal_info,
-}) => {
-    const icon_type = `${contract_type}${/^(call|put)$/i.test(contract_type) && barrier_count > 0 ? '_barrier' : ''}`.toLowerCase();
+class ContractInfo extends React.PureComponent {
+    constructor(props) {
+        super(props);
 
-    return (
-        <div className='box'>
-            <div className='left-column'>
-                <div className='type-wrapper'>
-                    <IconTradeType type={icon_type} className='type' />
-                </div>
-                <h4 className='trade-type'>{contract_title}</h4>
-            </div>
-            {(proposal_info.has_error || !proposal_info.id) ?
-                <div className={proposal_info.has_error ? 'error-info-wrapper' : ''}>
-                    <span>{proposal_info.message}</span>
-                </div>
-                :
-                <div className='purchase-info-wrapper'>
-                    <span className='purchase-tooltip'>
-                        <Tooltip alignment='left' icon='info' message={proposal_info.message} />
-                    </span>
-                    <div className='info-wrapper'>
-                        <div>{localize('Stake')}:</div>
-                        <div><Money amount={proposal_info.stake} currency={currency} /></div>
-                    </div>
-                    <div className='info-wrapper'>
-                        <div>{localize('Payout')}:</div>
-                        <div><Money amount={proposal_info.payout} currency={currency} /></div>
-                    </div>
-                    <div className='info-wrapper'>
-                        <div>{localize('Net Profit')}:</div>
-                        <div><Money amount={proposal_info.profit} currency={currency} /></div>
+        this.previous_price = null;
+    }
+
+    getIconType = () => {
+        let icon_type = '';
```

Comment:
it's better to check changes in props using React's `getDerivedStateFromProps` method, then compare the state of `previous_price` with the new price and do update on the states using `setState` accordingly

Reply:
I'm not sure if it's a good idea, as it's mentioned to better not use it by react,
https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html
``A common misconception is that getDerivedStateFromProps and componentWillReceiveProps are only called when props “change”. These lifecycles are called any time a parent component rerenders, regardless of whether the props are “different” from before. Because of this, it has always been unsafe to unconditionally override state using either of these lifecycles. Doing so will cause state updates to be lost.``.
why is that we need to use it?

Reply:
we want to show green icon for rise and red for fall of price and when the price is same we should have no icon, there are cases that getDerivedStateFromProps fires and price is not changed yet, and it goes in the condition of having same prev and current price, and the icon will be removed although really we didn't have change in proposal_info

Reply:
thanks, i was thinking since we are getting price whenever the `proposal_info` prop changes, i thought it's better to check for a change in props, and hence suggesting the use of `getDerivedStateFromProps` method

Reply:
with the latest changes you made, it seems to work fine 😄